### DISCUSSION TITLE
[2022 PROJECTS TURN-IN] Final Projects discussions and links

### DISCUSSION THREAD

**Discussion by williamgilpin** [williamgilpin/cphy#6] [archived]
Please post names, groups, topic areas, and eventually links to final projects in this thread. 

Once you have your repo name and URL settled, you can go ahead and edit your post to add it here even if it's unfinished (I won't grade it until the deadline in December). If you later change the URL, please edit the original post so that I don't get confused.

**Comment by DeepeshBVerma** [comment]
Deepesh Verma: optimization of python script that tests Simplified Kinetic Model (SKiM) plasma theory and comparison with compiled Fortran code. Implementation of jax, numba, or pytorch.

**Reply by williamgilpin** [reply]
imo jax is the most future-proof choice, for now. Julia could also make sense, but I have not used it enough to know (and it's an entirely different language). One thing to try might be to use "profilers" (ipython has a few built-in) to figure out what parts of the python code are slow---is there a for-loop than can be vectorized, or is it something like numerical integration where we need the output of the previous step to compute the next step? If the latter, speeding up whatever is called inside the loop could make sense

If there's numerical integration involved, some of my grad students have had huge success using [diffrax](https://docs.kidger.site/diffrax/), a jax library for numerical integration of ODEs

**Reply by leviyevalex** [reply]
If you need a powerful profiler, I've used the following before and its great:

https://github.com/joerick/pyinstrument

**Reply by DeepeshBVerma** [reply]
oop 
https://github.com/DeepeshBVerma/skim_phy381c.git

**Comment by jiangzz-lab** [comment]
Team members: Zhanzhi Jiang, Qiang Gao
Project title: Grover Method for Quantum Reinforcement Learning (QRL)
Abstract: We implemented a QRL agent based on the Grover algorithm and demonstrated its feasibility on simple environments
Project repository: https://github.com/jiangzz-lab/GroverQLearning 

Reference: 
[1] A qiskit-based quantum agent to play the quantum Tic-Tac-Toe game: https://github.com/ZhengtongYan/QRL
[2] Ganger, M. and Hu, W. International Journal of Intelligence Science, 9, 1-22 (2019)
[3] V. Saggio et al, Nature 591, 229 (2021)

**Reply by williamgilpin** [reply]
i don't know that much about QRL, but I would be very interested to read more. I'm not sure if it's still a common benchmark suite, but OpenAI's [gym](https://github.com/openai/gym) just got a refresh. Is the idea with QRL methods is that they match classical on classical hardware, but would have speedups on quantum circuits? Is the speedup on training or inference?

**Reply by jiangzz-lab** [reply]
To my knowledge, the experiments that have demonstrated speeding-up over classic approaches all rely on some sort of quantum hardware. The work in [2] is the only well-proved experiment that speeds up the learning process. They used the superposition of photons to shorten the time of feedback. There are a few other works that demonstrate the speeding up of deliberation time: https://iopscience.iop.org/article/10.1088/2058-9565/aaef5e

**Comment by liuyuezhang** [comment]
## Two-dimensional pattern formation in the Convolution-ReLU system
- Group member: Liu, Yuezhang.
### Reference
1. [Turing's model for biological pattern formation and the robustness problem](https://royalsocietypublishing.org/doi/10.1098/rsfs.2011.0113)
2. [Dynamics of pattern formation in lateral-inhibition type neural fields](https://link.springer.com/article/10.1007/BF00337259)
3. [A unified theory for the origin of grid cells through the lens of pattern formation](https://ganguli-gang.stanford.edu/pdf/19.DecodePattern.pdf)

**Reply by williamgilpin** [reply]
Check out chapter 2 of volume 2 of Murray: [diffusion driven instability](http://pcleon.if.ufrgs.br/pub/listas-sistdin/MurrayII.pdf), particularly Fig 2.4a. Perhaps there is a way to empirically find a "dispersion relation" like Fig 2.7 or parameter space like Fig 2.12? I think the width of the convolutional kernel is a proxy for the Sqrt(diffusivity) in Murray's notation

**Comment by PatrickGio** [comment]
Team: Patrick Giolando
Project: Utilize PIV algorithms for the quantification of 3D spatial vector fields of large motion and deformation problems, with the application to lung tissue.

**Reply by williamgilpin** [reply]
might be worthing taking a look into optical flow algos. i'm not sure when people use them vs PIV: https://en.wikipedia.org/wiki/Optical_flow

I could not find anything great for large-deformation PIV except this recent paper, which I'm not sure I follow:
https://ieeexplore.ieee.org/abstract/document/9638461

One possibility is that the innards of the central loop of PIV (patchwise cross correlation) can probably be refined somehow. For example, instead of just calculated mean displacement between two patches, fitting a local linear transform with bias (which would get mean translation + (skew,rot,scale)) might yield something interesting

**Comment by adamscottc** [comment]
### Ikeda-based nonlinear delayed dynamics for chaotic encryption of optical transmissions
Group Members: Inhwan Kim, Casey Christian, Adam Christensen

**References**

1. [Ikeda-based nonlinear delayed dynamics for application
to secure optical transmission systems using chaos
](https://github.com/williamgilpin/cphy/files/9636366/IKEDA_Larger2004.pdf)
2. [Quantitative universality for a class of nonlinear transformations](https://github.com/williamgilpin/cphy/files/9636377/Quantitative_universality_for_a_class_of_nonlinear.pdf)
3. [Chaotic attractors of an infinite-dimensional dynamical system](https://github.com/williamgilpin/cphy/files/9636496/10.1.1.558.2753.pdf)

**Reply by williamgilpin** [reply]
https://github.com/CSchoel/nolds

**Reply by williamgilpin** [reply]
http://www.physics.emory.edu/faculty/weeks//research/tseries1.html
http://www.physics.emory.edu/faculty/weeks//research/tseries7.html

**Reply by williamgilpin** [reply]
btw here's my old ikeda implementation. the @ decorators are because i used numba for hardware acceleration, but that's probably not necessary
https://github.com/williamgilpin/dysts/blob/master/dysts/maps.py

i used a chaotic fluid flow for cryptography a few years ago, it went okay
https://www.pnas.org/doi/10.1073/pnas.1721852115

**Reply by ihkim94** [reply]
Hi William, this is our final repo
[https://github.com/ihkim94/ikeda](https://github.com/ihkim94/ikeda)

Thank you for the lecture this semester. I have realized I have a lot of things to learn during my Ph. D.

**Comment by swilliamson7** [comment]
Team: Sarah Williamson

Project: Data-assimilation and reconstruction of parameters in a chaotic system using an adjusted Lagrange multiplier method and machine learning. I'll likely be using Julia for this. 

A few resources that discuss the general idea of what I'll try to do: 

1. [Controllability, not chaos, key criterion for ocean state estimation](https://npg.copernicus.org/articles/24/351/2017/) 
2. [Combining data assimilation and machine learning to emulate a dynamical model from sparse and noisy observations: A case study with the Lorenz 96 model](https://www.sciencedirect.com/science/article/pii/S1877750320304725?via%3Dihub)

**Reply by williamgilpin** [reply]
the famous neural-ODEs paper: adjoint method is a continuous limit of backprop
https://arxiv.org/abs/1806.07366
looks like they got it working in julia:
https://julialang.org/blog/2019/01/fluxdiffeq/

for ocean state estimation, you might find the LCS/FTLE literature interesting. a member of my group has been looking into calculating FTLE with nODEs, which might help discover transient gyres etc better than linear methods
http://georgehaller.com/reprints/annurev-fluid-010313-141322.pdf

**Reply by swilliamson7** [reply]
https://github.com/swilliamson7/chaos_predictions

**Comment by williamgilpin** [comment]
Michelle, Dominik, Irakli project on boosted forests [here](https://github.com/williamgilpin/cphy/discussions/14)

**Reply by micontre** [reply]
Hi William, 
Here is the repo for our final project :) 
https://github.com/micontre/boosted_random_forest

**Comment by williamgilpin** [comment]
Dalton project on Cherenkov & DTW [here](https://github.com/williamgilpin/cphy/discussions/17)

**Reply by engineer9001** [reply]
https://github.com/engineer9001/RootWaveformAnalyzer

**Comment by williamgilpin** [comment]
Hector Iglesias --- ringdown time series

**Reply by maxwellsdemon1965** [reply]
Here is my final repo:
[https://github.com/maxwellsdemon1965/cphy-ringdown-final-project](https://github.com/maxwellsdemon1965/cphy-ringdown-final-project)

**Comment by williamgilpin** [comment]
Jake McGrath -- active matter paper

**Reply by jakemcgrath1999** [reply]
https://github.com/jakemcgrath1999/muscle_model

here is my final repo!  I added a readme with different sources to remind you of the problem.  I put two helpful demos in there that explain how to run both the activation dynamics and the contractile dynamics code! 

Let me know if you have any questions, and thanks for the great semester!

**Comment by Tariq-Mahassen** [comment]
https://github.com/Tariq-Mahassen/PHY-381C/tree/main

Tariq Mahassen, Analysing Chaotic systems using the driven-damped pendulum

**Comment by ramreddy-physics** [comment]
https://github.com/ramreddy-physics/Madgraph_Search

Parameter Estimation on Collider data using DNNs
Ram Reddy

**Comment by liuyuezhang** [comment]
https://github.com/liuyuezhang/nsi

**Reply by liuyuezhang** [reply]
Sorry, I thought the DDL was on Dec 8th, but here we go.
